### PR TITLE
Parser - PWN_SAST Duplication Detection Change

### PR DESCRIPTION
**Pull request by ninp0**
**Description**

Change the duplicate detection logic (i.e. the line number may change even if the potentially vulnerable content does not).  This commit addresses that issue by relying upon a SHA256 value populated into unique_id_from_tool.

**Test results**

Unit test extended to ensure a unique_id_from_tool is generated.

**Checklist**

This checklist is for your information.

- [X ] Make sure to rebase your PR against the very latest `dev`.
- [ X] Features/Changes should be submitted against the `dev`.
- [ X] Bugfixes should be submitted against the `bugfix` branch.
- [X] Give a meaningful name to your PR, as it may end up being used in the release notes.
- [ X] Your code is flake8 compliant.
- [X ] Your code is python 3.11 compliant.
- [X ] If this is a new feature and not a bug fix, you've included the proper documentation in the docs at https://github.com/DefectDojo/django-DefectDojo/tree/dev/docs as part of this PR.
- [ X] Model changes must include the necessary migrations in the dojo/db_migrations folder.
- [ X] Add applicable tests to the unit tests.
- [ X] Add the proper label to categorize your PR.

### CODE REVIEW

File: dojo/settings/settings.dist.py [hunk 1]
```diff
@@ -1250,7 +1250,7 @@ def saml2_attrib_map_format(dict):
     'StackHawk HawkScan': ['vuln_id_from_tool', 'component_name', 'component_version'],
     'Hydra Scan': ['title', 'description'],
     'DrHeader JSON Importer': ['title', 'description'],
-    'PWN SAST': ['title', 'description'],
+    'PWN SAST': ['unique_id_from_tool'],
```
Comment (manuel-sommer) [inline dojo/settings/settings.dist.py:1253]: You also want to swith from DEDUPE_ALGO_HASH_CODE to DEDUPE_ALGO_UNIQUE_ID_FROM_TOOL

**Review by manuel-sommer — CHANGES_REQUESTED** [review]

File: dojo/settings/settings.dist.py (hunk 1 above)
Comment (ninp0) [reply dojo/settings/settings.dist.py:1253]: All set.  Thanks @manuel-sommer

Comment (manuel-sommer) [inline dojo/settings/settings.dist.py:1253]: I guess you can remove the line 'PWN SAST': ['unique_id_from_tool'], 

I made a PR regarding this unnecessary setting: https://github.com/DefectDojo/django-DefectDojo/pull/9188

**Review by manuel-sommer — CHANGES_REQUESTED** [review]

File: dojo/settings/settings.dist.py (hunk 1 above)
Comment (ninp0) [reply dojo/settings/settings.dist.py]: All set.  Thanks @manuel-sommer

**Review by mtesauro — APPROVED** [review]
Approved

**Review by cneill — APPROVED** [review]

**Review by grendel513 — APPROVED** [review]

**Review by Maffooch — APPROVED** [review]

**Pull request merged**